packages/react-hooks/src/useResource/useResource.ts

```diff
@@ -49,6 +48,8 @@ export function useResource<T extends Resource>(
             }
           }
         });
+    } else {
+      setResourceIfChanged(newValue);
     }
 
     return (() => (subscribed = false)) as () => void;
```

Comment:
why is resource in the dependency array here in useEffect? It makes sense this should depend on value. Reference is not referenced in the useEffect body

Reply:
Good catch, @yang-seen! Pushing out a followup PR to remove it.